applications/StructuralMechanicsApplication/custom_conditions/displacement_shifted_boundary_condition.cpp

```diff
@@ -0,0 +1,540 @@
+    StructuralMechanicsElementUtilities::CalculateB(*this, r_DN_DX, B_mat);
+    Vector strain_vect = prod(B_mat, unknown_values);
+
+    ConstitutiveLaw::Parameters cons_law_values(this->GetGeometry(), this->GetProperties(), rCurrentProcessInfo);
+    cons_law_values.SetShapeFunctionsValues(r_N);
+    cons_law_values.SetStressVector(stress_vect);
+    cons_law_values.SetStrainVector(strain_vect);
+    cons_law_values.SetConstitutiveMatrix(C_mat);
+
+    auto& r_cons_law_options = cons_law_values.GetOptions();
+    r_cons_law_options.Set(ConstitutiveLaw::COMPUTE_STRESS, false);
+    r_cons_law_options.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, true);
+    r_cons_law_options.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
+
+    p_cons_law->CalculateMaterialResponseCauchy(cons_law_values);
+
+    // Get Dirichlet BC imposition data
+    const double h = GetValue(ELEMENT_H);
+    const auto& r_bc_val = GetValue(DISPLACEMENT);
+    const double gamma = rCurrentProcessInfo[PENALTY_COEFFICIENT];
+
+    // Calculate the Nitsche BC imposition contribution
+    // 1. Add Nitsche penalty term
+    double aux_1;
+    double aux_2;
+    const double rho_C = norm_frobenius(C_mat); //TODO: GS uses the spectral radius in here
+    const double aux_weight = w * gamma * rho_C / h;
+    for (std::size_t i_node = 0; i_node < n_nodes; ++i_node) {
+        aux_1 = aux_weight * r_N[i_node];
+        for (std::size_t d = 0; d < n_dim; ++d) {
+            for (std::size_t j_node = 0; j_node < n_nodes; ++j_node) {
+                aux_2 = aux_1 * r_N[j_node];
+                rRightHandSideVector(i_node*n_dim + d) -= aux_2 * unknown_values(j_node*n_dim + d);
+            }
+            rRightHandSideVector(i_node*n_dim + d) += aux_1 * r_bc_val[d];
+        }
+    }
+
+    // 2. Add Nitsche stabilization term
+    Matrix aux_N(n_dim, local_size);
+    Matrix transB_C_proj(local_size, n_dim);
+    CalculateAuxShapeFunctionsMatrix(r_N, aux_N);
+    CalculateBtransCProjectionLinearisation(C_mat, B_mat, normal, transB_C_proj);
+    const Matrix stab_lhs = prod(transB_C_proj, aux_N);
+    Vector stab_rhs_bc = ZeroVector(local_size);
+    for (IndexType i = 0; i < local_size; ++i) {
+        for (IndexType d = 0; d < n_dim; ++d) {
+            stab_rhs_bc[i] += transB_C_proj(i,d) * r_bc_val[d];
+        }
+    }
+    const Vector stab_rhs_unk = prod(stab_lhs, unknown_values);
+    rRightHandSideVector += w * stab_rhs_bc;
+    rRightHandSideVector -= w * stab_rhs_unk;
+    KRATOS_CATCH("")
+}
+
+int DisplacementShiftedBoundaryCondition::Check(const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY
+
+    // Note that the base check is intentionally not called to avoid calling the base geometry check
+    return 0;
+
+    KRATOS_CATCH("")
+}
+
+void DisplacementShiftedBoundaryCondition::CalculateBtransCProjectionLinearisation(
+    const Matrix &rC,
+    const Matrix &rB,
+    const array_1d<double, 3> &rUnitNormal,
+    Matrix& rAuxMat)
```

Comment:
```suggestion
    const Matrix& rC,
    const Matrix& rB,
    const array_1d<double, 3>& rUnitNormal,
    Matrix& rAuxMat)
```
Inconsistent side of the &